### PR TITLE
Fix links array structure

### PR DESCRIPTION
**Pull request by darrenoh**

### CODE REVIEW

File: src/functions/theme.php [hunk 1]
```diff
@@ -11,6 +11,9 @@
 function theme(string $hook, array $variables = []): MarkupInterface|string
 {
     $build['#theme'] = $hook;
+    if (function_exists('retrofit_preprocess_' . $hook)) {
+        ('retrofit_preprocess_' . $hook)($variables);
+    }
```
Comment (mglaman) [inline src/functions/theme.php:16]: We're decorating the theme registry, is there a way to just do this there? It's how we register legacy "process" hooks.

**Comment by mglaman** [comment]
Actually, I also forgot about this:

```php
            if (method_exists(HookPreprocess::class, $theme_hook)) {
                $cache[$theme_hook]['preprocess functions'][] = HookPreprocess::class . '::' . $theme_hook;
            }
```

So we can move the function into that class.

File: src/functions/theme.php (hunk 1 above)
Comment (mglaman) [reply src/functions/theme.php:16]: This was moved, but I forgot to delete this.

File: src/Theme/HookPreprocess.php [hunk 2]
```diff
@@ -55,4 +55,39 @@ public static function block(array &$variables): void
...
+                    unset($link['href']);
+                }
+            }
+        }
+        if (!empty($variables['heading'])) {
+            if (!empty($variables['heading']['class'])) {
+                $variables['heading']['attributes']['class'] = $variables['heading']['class'];
+                unset($variables['heading']['class']);
+            }
+            $links['#heading'] = $variables['heading'];
```
[the remark this answers is not in the document]
Comment (darrenoh) [reply src/Theme/HookPreprocess.php:90]: Yes, that was leftover from an earlier attempt.

**Comment by darrenoh** [comment]
I'm having second thoughts about using a preprocess function on HookPreprocess. I don't want to modify every links array, just those that Drupal 7 modules pass to the theme() function.

**Review by mglaman — APPROVED** [review]

**Pull request merged**